pkg/k8s/watchers/service.go

```diff
@@ -576,10 +577,28 @@ func (k *K8sServiceWatcher) addK8sSVCs(svcID k8s.ServiceID, oldSvc, svc *k8s.Ser
 			} else {
 				scopedLog.WithError(err).Error("Error while inserting service in LB map")
 			}
+			k.updateServiceMappings(p)
 		}
 	}
 }
 
+// updateServiceMappings update mappings for the given service.
+// This is currently used for supporting high availability for kubeapi-server.
+func (k *K8sServiceWatcher) updateServiceMappings(svc *loadbalancer.SVC) {
+	if svc.Name.Name != "kubernetes" || svc.Name.Namespace != "default" {
+		return
+	}
+
+	var mapping client.K8sServiceEndpointMapping
+	mapping.Endpoints = make([]string, len(svc.Backends))
+	mapping.Service = svc.Frontend.L3n4Addr.AddrString()
+	for i := range mapping.Endpoints {
+		mapping.Endpoints[i] = svc.Backends[i].AddrString()
+	}
+	log.WithField("entry", mapping).Info("writing kubernetes service mapping")
+	_ = client.UpdateK8sAPIServerEntry(mapping)
```

Comment:
@dylandreimerink Perhaps a better way for the watcher to update the k8s client cell about these k8s api service mappings is via stateDB? Any comments?

Reply:
I don't think it makes much sense to create a table for just `K8sServiceEndpointMapping`, seeing as there is just 1 consumer and 1 producer.

You need to still make use of the service watcher to get the backends for a given server. However, with the experimental LB that is incoming, we should be able to let the `restConfigManager` directly watch the [services/frontends/backends tables](https://github.com/cilium/cilium/blob/main/pkg/loadbalancer/experimental/cell.go#L60).

It will still have to write to disk to persist it, but then its just one component, no communication over the FS anymore via file watchers.

Reply:
Thanks! Let's go ahead with the current approach in the meantime as discussed.